teuthology/openstack/__init__.py

```diff
@@ -963,7 +970,14 @@ def get_unassociated_floating_ip():
         """
         Return a floating IP address not associated with an instance or None.
         """
-        ips = json.loads(OpenStack().run("ip floating list -f json"))
+        try:
+            ips = json.loads(OpenStack().run("ip floating list -f json"))
+        except subprocess.CalledProcessError as e:
+            log.warning(e)
+            if e.returncode == 1:
+                return None
+            else:
+                raise e
```

Comment:
When it was just one line [1] being duplicated it wasn't a big deal. But now that you add this try..except block the amount of code being duplicated increases considerably. Can you split this out into a helper method to eliminate the code duplication?

[1] ips = json.loads(OpenStack().run("ip floating list -f json"))

Reply:
agree